### PR TITLE
Introduce Container component.

### PR DESCRIPTION
**Pull request by mattwagl**
- Introduce a new `Container` component as a replacement to `View`.
- Adjust `Application` to no longer use the View but a rather simple CSS grid as a base for SideBar based layouts.
- Remove `ApplicationServices` and move the functionality into `Application`.

### CODE REVIEW

**Comment by goloroden** [comment]
@vaalllaa @mattwagl There is a merge conflict, which is why this PR can not be merged. Can you please review this, and update this branch?

File: lib/components/Background.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export type Background = 'dark' | 'gray' | 'none';
```
Comment (goloroden) [inline lib/components/Background.ts]: I know that I was wondering about this already when I talked to @mattwagl a week ago, but now I can more explicitly tell what is causing trouble for me:

- `dark` sounds like a theme.
- `gray` is a color.
- `none` disables the background.

I'm fine with `none`, and I'm also fine with `dark` and `gray` on their own, but they don't fit very well together. They are things of two separate categories.

Should we maybe turn `dark` to `black` (or whatever is appropriate), or `gray` to `light` (or whatever feels right here)?

I'm not sure, but that's why I wanted to ask 😊

File: lib/components/Sidebar/README.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
 The `Sidebar` should be used to handle navigation when building an application layout using the `Application` component. Its also responsible for applying a consistent branding across the native web applications.
 
 ```jsx
-import { Application, Brand, Product, Sidebar, SidebarBrand, SidebarItem, SidebarFooter, View } from 'thenativeweb-ux';
+import { Application, Brand, Container, Product, Sidebar, SidebarBrand, SidebarItem, SidebarFooter } from 'thenativeweb-ux';
 
 <div style={{ position:'relative', width: 640, height: 480 }}>
-  <Application>
+  <Application useNotification={ false } useDialogs={ false }>
```
Comment (goloroden) [inline lib/components/Sidebar/README.md]: Shouldn't `useNotification` be `useNotifications`, in the same way as it's `useDialogs`?

File: lib/components/ThemeProvider/README.md [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
+  SidebarItem,
+  ThemeProvider,
+  SidebarFooter,
+  themes
 } from 'thenativeweb-ux';
 
 <div style={{ position: 'relative', width: 640, height: 480 }}>
   <ThemeProvider theme={ themes.wolkenkit }>
-    <Application>
+    <Application useNotification={ false } useDialogs={ false }>
```
Comment (goloroden) [inline lib/components/ThemeProvider/README.md]: Same here … plural form?

File: lib/components/Website/README.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 The `Website` component acts as your root component when  building a content-driven, grid-based communication website. It will make sure the most basic things like fonts are available to you.
```
Comment (goloroden) [inline lib/components/Website/README.md]: There is a space too much between `when` and `building`.

File: lib/components/Website/README.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
...
 
-If you're building an interaction-driven, flexbox-based appliction layout use the `Application` component as your root component instead.
+If you're building an interaction-driven application layout use the `Application` component as your root component instead.
```
Comment (goloroden) [inline lib/components/Website/README.md]: I think the word `layout` is wrong here, especially if you compare it with the paragraph before.

File: lib/components/layout/Container/styles.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+  BackgroundGray: {
+    background: theme.color.brand.grayLight
+  },
+
+  BackgroundNone: {
+    background: 'none'
+  },
+
+  // For all the horizontal and vertical positioning options we're not
```
Comment (goloroden) [inline lib/components/layout/Container/styles.ts]: Can you please align this with 80 columns?

File: test/integration/components/ContainerTests.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+suite('components/layout/Container', (): void => {
+  let page: Page;
+
+  teardown(async (): Promise<void> => {
+    await browser.teardownPage(page);
+  });
+
+  suite('isScrollable', (): void => {
+    test('causes the container to be scrollable when viewport is smaller than the parent.', async (): Promise<void> => {
```
Comment (goloroden) [inline test/integration/components/ContainerTests.ts]: `viewport` => `the viewport`

Comment (goloroden) [reply test/integration/components/ContainerTests.ts]: And there's something missing such as `... if set to true`.

Which also means there's a test missing for the opposite case.

File: lib/components/Sidebar/README.md (hunk 2 above)
Comment (vaalllaa) [reply lib/components/Sidebar/README.md]: Actually it is already useNotifications, here it's a typo. ✌️

File: lib/components/ThemeProvider/README.md (hunk 3 above)
Comment (vaalllaa) [reply lib/components/ThemeProvider/README.md]: yes, same typo here.

File: lib/components/Background.ts (hunk 1 above)
Comment (vaalllaa) [reply lib/components/Background.ts]: Makes totally sense. I switched from 'gray' to 'light'.

**Pull request merged**